cmake/ProjectVersion.cmake

```diff
@@ -87,18 +87,30 @@ ENDIF()
 
 set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_REV}.${PROJECT_VERSION_TWEAK}")
 
+# Define a constant length for the commit hash
+set(FW_COMMIT_HASH_LENGTH 9)
+
+# Create fallback value with constant length
+string(REPEAT "0" ${FW_COMMIT_HASH_LENGTH} FW_COMMIT_HASH_UNKNOWN)
```

Comment:
This is at odds with/will get clobbered by line 94-ish of CMakeLists.txt.

Reply:
You're right, I missed it. Will look into it.